### PR TITLE
Support "globalApi: false" in esm output

### PR DESCRIPTION
**Pull request by maxburs**
Update monaco-kusto to support apps that don't have monaco's global api enabled so our esm output can be used by Webpack.

A Webpack bug currently prevents monaco's globalApi from being enabled because of a bug in it's transpiling.

## Changes
* Update source files to resolve monaco via import of `monaco-editor/esm/vs/editor/editor.api` instead of a global variable.
  * `editor.api` just includes the parts we need. `monaco-editor` includes a lot of optional features and every language.
  * Got types working with a lot of casts with isn't ideal. This could be improved, but those improvements might not make sense for this PR. 
* Update AMD build scripts to convert monaco import into a global variable
* Remove "globalApi: true" config from Parcel ESM example
* Exported `monaco.languages.kusto` api from `monaco.contribution`. We're not able to modify the types (I think?) of monaco's non-global api. This new api gives consumers a type-safe version.
  * Updated Parcel example to use this.
* Update Parcel example to not build monaco workers for languages besides Kusto

### CODE REVIEW

File: package/scripts/lib.js [hunk 1]
```diff
@@ -114,6 +114,8 @@ export function rollupAMDOutput(type) {
         dir: path.join(packageFolder, 'release', type),
         sourcemap: !process.env.CI,
         plugins: [type === 'min' && terser()],
-        globals: Object.fromEntries(languageServiceFiles),
```
Comment (maxburs) [inline package/scripts/lib.js:117]: Unused. (Accidentally left over from when I was still trying to get the build system working)

File: package/src/kustoMode.ts [hunk 2]
```diff
@@ -21,16 +19,19 @@ let workerPromise: Promise<WorkerAccessor> = new Promise((resolve, reject) => {
  * Called when Kusto language is first needed (a model has the language set)
  * @param defaults
  */
-export function setupMode(defaults: LanguageServiceDefaultsImpl, monacoInstance: typeof monaco): WorkerAccessor {
+export function setupMode(
+    defaults: LanguageServiceDefaultsImpl,
+    monacoInstance: typeof globalThis.monaco
```
Comment (Weffe) [inline package/src/kustoMode.ts:24]:
```suggestion
    monacoInstance: monaco
```

I'm not too familiar with the types here but we can't reference the `monaco` type that's from the import statement on line 1 (e.g. `import * as monaco from 'monaco-editor'`?

Comment (Weffe) [reply package/src/kustoMode.ts:24]: Disregard.

**Review by Weffe — APPROVED** [review]
Nice changes here. The changes look really simple so I'm glad you figured out a way to work around the Webpack bug by just refactoring a bit on our side.

**Pull request merged**